protobuf/v1/nexus.proto

```diff
@@ -67,6 +70,18 @@ message CreateNexusRequest {
   NexusNvmePreemption preemptPolicy = 11; // the preemption policy to use
 }
 
+// Create nexus snapshot.
+message  CreateSnapshotRequest {
+  string nexus_uuid = 1;    // UUID of the nexus
+  string snapshot_name = 2; // User-specific snapshot name.
+  string txn_id = 3;        // unique transaction id generate for snapshot creation
+}
+
+// Nexus snapshot creation response.
+message CreateSnapshotResponse {
+  string snapshot_uuid = 1; // UUID of the nexus snapshot
```

Comment:
what's the difference between txn_id and snapshot_uuid here?

Reply:
Snapshot UUID is the unique UUID of the snapshot, like UUID of the device.
TX_ID is not supposed the be unique, it's part of the key (entity_id, tx_id), an this combination is supposed to be unique.
So we return UUID (which is generated by SPDK).

Reply:
do we need to also specify this uuid as an argument as well then so we can make the call idempotent? I think for the lvol for example we use methods with allow us to specify a UUID selected by the control-plane.

Reply:
OK, we could have that if that's needed to the Control Plane.

Reply:
For Replica Snapshot We are following the same as nexus.. We get the txn-id from control plane and return UUID from dataplane which is generated by SPDK.

// Create Replica Snapshot Request.
message CreateReplicaSnapshotRequest {
  string replica_uuid = 1;  // uuid of the replica
  string snapshot_name = 2; // snapshot to be created
  string txn_id = 3; // unique transaction id generate for snapshot creation
}

/// Create Replica Snapshot Response.
message CreateReplicaSnapshotResponse {
  string snapshot_uuid = 1; // UUID of the snapshot newly created
}